app/models/course/assessment/question.rb

```diff
@@ -79,6 +79,13 @@ def copy_attributes(other)
     self.maximum_grade = other.maximum_grade
   end
 
+  # Finds the submission_question of a question given the submission_id
+  #
+  # @return [Course::Assessment::SubmissionQuestion]
+  def submission_question_from_submission(submission_id)
```

Comment:
This can be an [association scope](http://guides.rubyonrails.org/association_basics.html#association-extensions).

Reply:
This returns a single `SubmissionQuestion` object though, shouldn't scopes normally be used if they return `ActiveRecord::Associations::CollectionProxy` instead so that we can chain them or get collections?

Reply:
Actually I this is a minor nitpick on my end - the scope can return a single object using `find_by, can see the example . Shifting it in just reduces the need to specify `submission_questions` at the start Of the next line.